frontend/packages/topology/src/data-transforms/DataModelProvider.tsx

```diff
@@ -10,6 +14,23 @@ interface DataModelProviderProps {
   children?: React.ReactNode;
 }
 
+export const getNamespacedDynamicModelFactories = (
+  factories: LoadedExtension<DynamicTopologyDataModelFactory>[],
+) =>
+  factories.map(({ properties, ...ext }) => ({
+    ...ext,
+    properties: {
+      ...properties,
+      resources: (ns: string) =>
+        Object.assign(
+          {},
+          ...Object.entries(properties.resources).map(([k, v]) => ({
+            [k]: { ...v, namespace: ns },
```

Comment:
We should probably accept the namespace provided before overriding? Not sure we have a use case for this yet but if an extension wanted to pull resources from a specific namespace we'd have to do:

```suggestion
            [k]: { namespace: ns, ...v },
```